src/js/utils/html-utils.js

```diff
@@ -6,41 +6,75 @@
  */
 (function(define) {
     'use strict';
-    define(['underscore', 'edx-ui-toolkit/js/utils/string-utils'], function(_, StringUtils) {
-        var escape, interpolateHtml, HTML, template;
+    define(['underscore', 'jquery', 'edx-ui-toolkit/js/utils/string-utils'], function(_, $, StringUtils) {
+        var ensureHtml, element, interpolateHtml, HTML, template, setHtml, append, prepend;
 
         /**
-         * Helper function to return a string flagged as an HTML snippet.
+         * Creates an HTML snippet.
+         *
+         * The intention of an HTML snippet is to capture a string that
+         * is known to contain HTML that has been safely escaped.
+         * As an example, this allows interpolate to understand that
+         * it does not need to further escape this HTML.
+         *
+         * @param {string} htmlString The string of HTML.
+         * @constructor
+         */
+        function HtmlSnippet(htmlString) {
+            this.text = htmlString;
+        }
+        HtmlSnippet.prototype.valueOf = function() {
+            return this.text;
+        };
+        HtmlSnippet.prototype.toString = function() {
+            return this.text;
+        };
+
+        /**
+         * Helper function to create an HTML snippet from a string.
          *
          * @param {string} htmlString The string of HTML.
-         * @returns {string} A string flagged as an HTML snippet.
+         * @returns {HtmlSnippet} An HTML snippet that can be safely rendered.
          * @constructor
          */
         HTML = function(htmlString) {
-            var htmlSnippet = new String(htmlString); // jshint ignore:line
-            htmlSnippet.isHtmlSnippet = true;
-            return htmlSnippet;
+            return new HtmlSnippet(htmlString);
         };
 
         /**
-         * Returns an HTML string that is appropriately escaped.
+         * Ensures that the specified HTML is encoded as an HtmlSnippet.
          *
-         * Note: if the string provided is flagged as being an HTML snippet
+         * If the HTML is provided as a string, then it is HTML-escaped and
+         * returned as an HtmlSnippet. If the parameter is an HTML snippet
          * then it will be returned directly so as not to double escape it.
          *
-         * @param {string} text A string that is either plan text, or is
-         * flagged as being an HTML snippet.
-         * @returns {string} A safely HTML-escaped string.
+         * @param {(string|HtmlSnippet)} html HTML provided either as
+         * a string or as an HtmlSnippet.
+         * @returns {HtmlSnippet} A safely escaped HTML snippet.
          */
-        escape = function(text) {
-            if (text.isHtmlSnippet) {
-                return text.toString();
+        ensureHtml = function(html) {
+            if (html instanceof HtmlSnippet) {
+                return html.toString();
```

Comment:
I don't understand the inconsistent return values of this method. In this case (html already an HtmlSnippet), this method returns a String. But in the case of a String being passed in, an HTMLSnippet is returned.